src/signtool/IBM_Container.cpp

```diff
@@ -0,0 +1,1767 @@
+        {
+            pUtils->ReadFromFile( p_keyFileName,
+                                  p_buffer,
+                                  ECDSA521_KEY_SIZE + 1 ); // since keyfile is 133 bytes with 0x04
+                                                           // at the begining
+
+            if (p_buffer[0] != g_UNCOMPRESSED_PUBKEY_FORMAT)
+            {
+                std::stringstream ss;
+                ss << "File <" 
+                   << p_keyFileName 
+                   << "> is not a vaid p521 public key file"
+                   << std::endl;
+
+                THROW_EXCEPTION_STR(ss.str().c_str());
+            }
+            
+            // public key file OK, delete the first byte
+            p_buffer.erase( p_buffer.begin() );
+        }
+        else if (p_mode == IBM_Utils::g_MODE_DEVELOPMENT)
+        {
+            pUtils->GetPublicKeyBytes( p_keyFileName.c_str(), p_buffer );
+        }
+        else
+        {
+            // should never get here
+            std::stringstream ss;
+            ss << "mode <" << p_mode << "> is not supported, must be one of "
+               << IBM_Utils::g_MODE_PRODUCTION << " or " << IBM_Utils::g_MODE_DEVELOPMENT;
+
+            THROW_EXCEPTION_STR(ss.str().c_str());
+        }
+
+        THROW_EXCEPTION(p_buffer.size() != ECDSA521_KEY_SIZE);
+    }
+
+
+    void ReadSignatureFromFile( const std::string&    p_mode,
+                                const std::string&    p_sigFileName,
+                                std::vector<uint8_t>& p_buffer )
+    {
+        IBM_Utils* pUtils = IBM_Utils::get();
+        THROW_EXCEPTION(pUtils == NULL);
+
+        if (p_mode == IBM_Utils::g_MODE_PRODUCTION)
+        {
+            pUtils->ReadFromFile( p_sigFileName, p_buffer, ECDSA521_SIG_SIZE );
+        }
+        else if (p_mode == IBM_Utils::g_MODE_DEVELOPMENT)
+        {
+            pUtils->GetSignatureBytes( p_sigFileName.c_str(), p_buffer );
+        }
+        else
+        {
+            // should never get here
+            std::stringstream ss;
+            ss << "mode <" << p_mode << "> is not supported, must be one of "
+               << IBM_Utils::g_MODE_PRODUCTION << " or " << IBM_Utils::g_MODE_DEVELOPMENT;
+
+            THROW_EXCEPTION_STR(ss.str().c_str());
+        } 
+
+        THROW_EXCEPTION(p_buffer.size() != ECDSA521_SIG_SIZE);
+    } 
+}
+
+
+
+// ContainerHdr C'tor
+ContainerHdr::ContainerHdr()
+   : m_magicNumber( ROM_MAGIC_NUMBER ),
+     m_version( CONTAINER_VERSION ),
+     m_containerSize(0),
+     m_targetHrmor(0),
+     m_stackPointer(0),
+     m_hwPkeyA(),
+     m_hwPkeyB(),
+     m_hwPkeyC()
+{
+}
+
+
+/**
+ * @brief   Get the Container Header message as a byte stream
+ *
+ * @param[out]  packet  -  A reference to a vector of type uint8_t
+ */
+void ContainerHdr::GetHeaderBytes( std::vector<uint8_t>& packet ) const
+{
+    packet.clear();
+
+    // insert the magic nummber field
+    packet.push_back((m_magicNumber >> 24) & 0xFF);
```

Comment:
I suspect these could all be significantly cleaner with an insert call from a std::array or initializer_list.

```
auto getByte = [](auto value, auto byte) { return (value >> (byte*8)) & 0xff; }

packet.insert(packet.end(), { getByte(m_magicNumber, 3), getByte(m_magicNumber, 2) ... });
```